File: templates/templates/FSharp.EpoxyHello.Wpf/EpoxyHello/ViewModels/MainWindowViewModel.fs
namespace EpoxyHello.ViewModels

open Epoxy
open System

open EpoxyHello.Models

[<Sealed; ViewModel>]
type public MainWindowViewModel() as self =
    do
        // A handler for window loaded
        self.Ready <- Command.Factory.create(fun () -> async {
           self.Title <- "Hello Epoxy!"
        })
 
    member val Ready: Command = null with get, set
    member val Title: string = "" with get, set

File: templates/templates/EpoxyHello.Wpf/EpoxyHello/ViewModels/MainWindowViewModel.cs
﻿using Epoxy;
using System;

namespace EpoxyHello.ViewModels;

[ViewModel]
public sealed class MainWindowViewModel
{
    public Command Ready { get; }
    public string Title { get; private set; } = "";

    public MainWindowViewModel()
    {
        // A handler for window loaded
        this.Ready = Command.Factory.Create(() =>
        {
            this.Title = "Hello Epoxy!";
            return default;
        });
    }
}
